src/SpeedyTransforms/spectral_transform.jl

```diff
@@ -143,38 +146,48 @@ function SpectralTransform(
     end
     
     # SCRATCH MEMORY FOR FOURIER NOT YET LEGENDRE TRANSFORMED AND VICE VERSA
-    scratch_memory_north = zeros(Complex{NF}, nfreq_max, nlayers, nlat_half)
-    scratch_memory_south = zeros(Complex{NF}, nfreq_max, nlayers, nlat_half)
+    # convert all to the arraytype (e.g. moves array to GPU)
+    scratch_memory_north = ArrayType_(zeros(Complex{NF}, nfreq_max, nlayers, nlat_half))
+    scratch_memory_south = ArrayType_(zeros(Complex{NF}, nfreq_max, nlayers, nlat_half))
 
     # SCRATCH MEMORY TO 1-STRIDE DATA FOR FFTs
-    scratch_memory_grid  = zeros(NF, nlon_max*nlayers)
-    scratch_memory_spec  = zeros(Complex{NF}, nfreq_max*nlayers)
+    scratch_memory_grid  = ArrayType_(zeros(NF, nlon_max*nlayers))
+    scratch_memory_spec  = ArrayType_(zeros(Complex{NF}, nfreq_max*nlayers))
 
     # SCRATCH MEMORY COLUMNS FOR VERTICALLY BATCHED LEGENDRE TRANSFORM
-    scratch_memory_column_north = zeros(Complex{NF}, nlayers)
-    scratch_memory_column_south = zeros(Complex{NF}, nlayers)
+    scratch_memory_column_north = ArrayType_(zeros(Complex{NF}, nlayers))
+    scratch_memory_column_south = ArrayType_(zeros(Complex{NF}, nlayers))
 
-    # PLAN THE FFTs
-    FFT_package = NF <: Union{Float32, Float64} ? FFTW : GenericFFT
     rfft_plans = Vector{AbstractFFTs.Plan}(undef, nlat_half)
     brfft_plans = Vector{AbstractFFTs.Plan}(undef, nlat_half)
     rfft_plans_1D = Vector{AbstractFFTs.Plan}(undef, nlat_half)
     brfft_plans_1D = Vector{AbstractFFTs.Plan}(undef, nlat_half)
 
-    fake_grid_data = zeros(Grid{NF}, nlat_half, nlayers)
-
-    for (j, nlon) in enumerate(nlons)
-        real_matrix_input = view(fake_grid_data.data, rings[j], :)
-        complex_matrix_input = view(scratch_memory_north, 1:nlon÷2 + 1, :, j)
-        real_vector_input = view(fake_grid_data.data, rings[j], 1)
-        complex_vector_input = view(scratch_memory_north, 1:nlon÷2 + 1, 1, j)
+    fake_grid_data = adapt(ArrayType_, zeros(Grid{NF}, nlat_half, nlayers))
 
-        rfft_plans[j] = FFT_package.plan_rfft(real_matrix_input, 1)
-        brfft_plans[j] = FFT_package.plan_brfft(complex_matrix_input, nlon, 1)
-        rfft_plans_1D[j] = FFT_package.plan_rfft(real_vector_input, 1)
-        brfft_plans_1D[j] = FFT_package.plan_brfft(complex_vector_input, nlon, 1)
-    end
+    # PLAN THE FFTs
+    plan_FFTs!(
+        rfft_plans, brfft_plans, rfft_plans_1D, brfft_plans_1D,
+        fake_grid_data, scratch_memory_north, rings, nlons
+    )
     
+    # PRECOMPUTE KJM INDICES FOR LEGENDRE TRANSFORM (0-based)
```

Comment:
Can we slightly more explicit in the documentation/comments what variables like `kjm` stands for? I was confused at first, but it's 
layer - ring - m (SPH number), right?

Reply:
that's usually the indexing I've been using, `i` is the in-ring index (i.e. longitude), `j` is the ring index (latitude), `ij` is the running index for all grid points across rings, `l` is the spherical harmonics degree (1st dimension, row), `m` is the order (2nd dimension, column), `k` is for both grid and spherical harmonics any additional dimensions, i.e. vertical layer mostly.

Reply:
Oh and `lm` is consistently a running index across all spherical harmonics ...